tests/ui/features/bootstrap/TextEditorContext.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+* ownCloud
+*
+* @author Phillip Davis
+* @copyright 2017 Phillip Davis info@individual-it.net
+*
+* This library is free software; you can redistribute it and/or
+* modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or any later version.
+*
+* This library is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+*
+* You should have received a copy of the GNU Affero General Public
+* License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+*
+*/
+
+use Behat\Behat\Context\Context;
+use Behat\Behat\Hook\Scope\BeforeScenarioScope;
+use Behat\Gherkin\Node\PyStringNode;
+use Behat\MinkExtension\Context\RawMinkContext;
+use Page\TextEditorPage;
+
+require_once 'bootstrap.php';
+
+/**
+ * Files context.
+ */
+class TextEditorContext extends RawMinkContext implements Context
+{
+	private $textEditorPage;
+	private $featureContext;
+	private $filesContext;
+
+	public function __construct(TextEditorPage $textEditorPage)
+	{
+		$this->textEditorPage = $textEditorPage;
+	}
+
+	/**
+	 * @When /^I create a text file with the name ((?:'[^']*')|(?:"[^"]*"))( and the default file type|)$/
```

Comment:
I want ( and the default file type) to be an optional phrase on the end of the step text. So for that I think I have to dive in regex land.
But the name of the file could be in single or double quotes, so I have to look for either a group surrounding by single quotes, or a group surrounded by double quotes.
The regex then captures the whole thing, including the outside pair of quotes.
There are alternative regex that first capture the which quote it is:
```
(["'])
```
and then use that with a back-reference to the captured quote.

But then the quote comes through as the first capture group, and ends up as a parameter to the function!

If if match the first quote with a non-capturing group, then I cannot use a back-reference to it:
```
(?:["'])
```
So what I have done here is the best I have come up with so far.

Reply:
why do you need `and the default file type`? what other file types could a text file have? Do you mean the file extension `.txt`?

Reply:
The "new text file name" box comes up with "New text file.txt" and "New text file" is highlighted. If you just type straight away, without clearing the field, then it retains the ".txt". So I want to test that behavior, using "the default file type" that is presented by default.
Is it easier to understand if I call it "the default file extension?

Reply:
"the default file extension" or "without changing the default file extension"